### PR TITLE
Handle W.lastRenderedPageBreak in UnicodeMapper

### PR DESCRIPTION
**Pull request by luizfbicalho**
I did this assembly using the old OpenXmlPowerTools and it worked fine (Except from the > in the other PR), but when I do the same assembly in the clippit I get an error on save

  Assert.Fail(): '', hexadecimal value 0x01, is an invalid character.

I created a test to show this error and get some insight on why this is happening.

[result.docx](https://github.com/sergey-tihon/Clippit/files/9575668/result.docx)
[data.txt](https://github.com/sergey-tihon/Clippit/files/9575669/data.txt)
[input.docx](https://github.com/sergey-tihon/Clippit/files/9575670/input.docx)

### CODE REVIEW

File: OpenXmlPowerTools.Tests/Word/DocumentAssemblerTests.cs [hunk 1]
```diff
@@ -206,6 +207,42 @@ public void DA102_Throws(string name, string data)
...
+        {
+            DA101(name, data, err);
+            var assembledDocx = new FileInfo(Path.Combine(TempDir, name.Replace(".docx", "-processed-by-DocumentAssembler.docx")));
+            var afterAssembling = new WmlDocument(assembledDocx.FullName);
+
+            var descendants = afterAssembling.MainDocumentPart.Value;
+
+            if (descendants.Contains(">"))
+            {
+                Assert.Fail("Found > on text");
```
Comment (sergey-tihon) [inline OpenXmlPowerTools.Tests/Word/DocumentAssemblerTests.cs]: There is not such thing as `Asset.Fail`, can you please make it compilable?

Comment (luizfbicalho) [reply OpenXmlPowerTools.Tests/Word/DocumentAssemblerTests.cs]: I updated to xunit 2.4.2, now I changed back to 2.4.1

**Comment by sergey-tihon** [comment]
I believe that the issue in `DA-XmlError.docx`
Some bad characters inside

image

**Comment by luizfbicalho** [comment]
> I believe that the issue in `DA-XmlError.docx` Some bad characters inside
> 
> image

I understand that something happened or in the document or in the xml, but the real issue is that this doesn't happen in the old OpenXmlPowerTools

If you get the word part <#<Content Select="translate(./Conjuge/Email,'ABCDEFGHIJKLMNOPQRSTUVWXYZÂÁÀÃÄÊÈÉËÎÌÍÏÕÔÒÓÖÜÛÚÙÑ','abcdefghijklmnopqrstuvwxyzâáàãäêèéëîìíïõôòóöüûúùñ')"/> #> there is nothing between the öü, and if you get in debugger there is this öü 

I think that this is something related to page breaks or line breaks inside the xml of the content, shoudn't It be treated in the line  512 to remove from the regex?

**Comment by luizfbicalho** [comment]
I Added a Xml Cleaning method

**Review by luizfbicalho — COMMENTED** [review]
update to validate the xml and remove invalid characters

**Review by luizfbicalho — COMMENTED** [review]
change case in file to correct test in linux

**Review by luizfbicalho — COMMENTED** [review]
update xml validation to use the .net xmlconvert.isxmlchar and correct casing in linux

**Review by luizfbicalho — COMMENTED** [review]
updaet case of file for linux

File: OpenXmlPowerTools/Word/DocumentAssembler.cs [hunk 2]
```diff
@@ -514,7 +514,8 @@ private static object TransformToMetadata(XNode node, TemplateError te)
                     OpenXmlRegex.Replace(new[] { element }, r, thisGuid, (_, match) =>
                     {
                         var matchString = match.Value.Trim();
-                        var xmlText = matchString.Substring(2, matchString.Length - 4).Trim().Replace('“', '"').Replace('”', '"');
+                        var xmlText = CleanInvalidXmlChars(matchString.Substring(2, matchString.Length - 4).Trim().Replace('“', '"').Replace('”', '"'));
```
Comment (sergey-tihon) [inline OpenXmlPowerTools/Word/DocumentAssembler.cs]: @luizfbicalho I am not sure that DocumentAssembler actually should replace invalid XML characters.

Imagine for a moment that `DocumentAssemblier` (sort of interpretator) , that parse `*.docx` (kind of source code)  and run instructions against XML (data). What do you expect from the interpretator when you send an invalid source code for execution? An error that the source program is incorrect with position details or some magic attempt inside the code to auto-correct the source and potentially invalid/unexpected execution result?

I believe that when sourcing (`*.docx`) contains invalid characters inside XML block, we should report an error with instructions on how to find it.

Comment (luizfbicalho) [reply OpenXmlPowerTools/Word/DocumentAssembler.cs]: The problem is not in what you send, the problem is in what word puts, look at this case that is the example, the unicode invalid char isn't in the word document, it's probably some line break or page break that turns into this invalid char

I understand that if I put something wrong the assembler should not correct, but the word format breaks the xml and this is what I'm correcting

**Comment by sergey-tihon** [comment]
@luizfbicalho plz check that latest changes here works for you.

**Comment by luizfbicalho** [comment]
> @luizfbicalho plz check that latest changes here works for you.

how do I get this back to my repo?

**Comment by sergey-tihon** [comment]
Just pull, it is already in your fork

**Review by luizfbicalho — COMMENTED** [review]
This solved the > problem

**Review by luizfbicalho — COMMENTED** [review]
This solved the Unicode Error on XML

**Comment by luizfbicalho** [comment]
> @luizfbicalho plz check that latest changes here works for you.

Looks like its all working, I'll try to update my libraries later, thanks

**Pull request merged**